enginetest/insert_queries.go

```diff
@@ -441,6 +458,186 @@ var InsertScripts = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "auto increment on tinyint",
+		SetUpScript: []string{
+			"create table auto (pk tinyint primary key auto_increment)",
+			"insert into auto values (NULL),(10),(0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "select * from auto",
+				Expected: []sql.Row{
+					{1},{10},{11},
+				},
+			},
+		},
+	},
+	{
+		Name: "auto increment on smallint",
+		SetUpScript: []string{
+			"create table auto (pk smallint primary key auto_increment)",
+			"insert into auto values (NULL),(10),(0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "select * from auto",
+				Expected: []sql.Row{
+					{1},{10},{11},
+				},
+			},
+		},
+	},
+	{
+		Name: "auto increment on mediumint",
+		SetUpScript: []string{
+			"create table auto (pk mediumint primary key auto_increment)",
+			"insert into auto values (NULL),(10),(0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "select * from auto",
+				Expected: []sql.Row{
+					{1},{10},{11},
+				},
+			},
+		},
+	},
+	{
+		Name: "auto increment on int",
+		SetUpScript: []string{
+			"create table auto (pk int primary key auto_increment)",
+			"insert into auto values (NULL),(10),(0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "select * from auto",
+				Expected: []sql.Row{
+					{1},{10},{11},
+				},
+			},
+		},
+	},
+	{
+		Name: "auto increment on bigint",
+		SetUpScript: []string{
+			"create table auto (pk bigint primary key auto_increment)",
+			"insert into auto values (NULL),(10),(0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "select * from auto",
+				Expected: []sql.Row{
+					{1},{10},{11},
+				},
+			},
+		},
+	},
+	{
+		Name: "auto increment on tinyint unsigned",
+		SetUpScript: []string{
+			"create table auto (pk tinyint unsigned primary key auto_increment)",
+			"insert into auto values (NULL),(10),(0)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "select * from auto",
+				Expected: []sql.Row{
+					{uint64(1)},{uint64(10)},{uint64(11)},
```

Comment:
Heads up that the current test implementation for row comparison widens all numeric types before comparison, so these casts are all no-ops. If you think this is important behavior to test you will have to come up with another way.